### PR TITLE
WIP - slide in the simulator from the right

### PR DESCRIPTION
**Pull request by thutterer**
OK, this is another attempt for the simulator placing. Other than #55 this one slides in from the right like @tboerger suggested.

__It's not done!__ I just want to show both of you how it looks so far.
I have not changed the content of simulator-panel much. You might dislike the general idea, so I will wait until I get a "go" for this approach. Like Thomas suggested, the toggle-Button should probably be on the right screen border and not in the top navbar... So don't nitpick yet on such thing (or the naming of the ids)

Also note that I use the jQuery library [bigSlide.js](http://ascott1.github.io/bigSlide.js) and that I had to change its source code a bit to get what we want. (only the CSS rules but still)

So do you like this (in general) better than #55? Personally I don't know. #55 is more mobile-friendly I guess and way simpler.

### CODE REVIEW

File: hawk/app/views/shared/_simulator.html.haml [hunk 1]
```diff
@@ -15,23 +15,22 @@
...
+            %button#sim-addop.btn.btn-default.btn-sm{type: :button}
+              = icon_text "plus", _("Operation")
+            - unless current_cib.tickets.empty?
+              %button#sim-addticket.btn.btn-default.btn-sm{type: :button}
+                = icon_text "plus", _("Ticket")
+            %button#sim-delete.btn.btn-default.btn-sm{type: :button}
+              = icon_tag "minus"
+          .row
+            .well
+              %button#sim-results.btn.btn-default{type: :button, disabled: :disabled}
```
Comment (thutterer) [inline hawk/app/views/shared/_simulator.html.haml]: > Classes should be listed before IDs (.btn should precede #sim-results)

Really? I dislike this rule.

File: hawk/app/assets/javascripts/application.js [hunk 2]
```diff
@@ -69,7 +70,9 @@ $(function() {
   );
 
   $(window).on(
-    'load',
-    createMutationObserver
+    'load', function(){
+      createMutationObserver();
+      $('.menu-link').bigSlide({side: 'right', menuWidth: '500px'});
```
Comment (tboerger) [inline hawk/app/assets/javascripts/application.js]: Why did you put it here? That looks totally wrong.

File: hawk/app/assets/javascripts/bigSlide.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+        // return a reference to the DOM selection bigSlide.js was called on
+        // so that the destroy method is chainable
+        return menuLink;
+      },
+
+      // update the menu's state
+      changeState: function(){
+        if (model.state === 'closed') {
+          model.state = 'open'
```
Comment (tboerger) [inline hawk/app/assets/javascripts/bigSlide.js]: That should be added to be able to minify the code.

File: hawk/app/assets/javascripts/bigSlide.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        } else {
+          model.state = 'closed'
```
Comment (tboerger) [inline hawk/app/assets/javascripts/bigSlide.js]: That should be added to be able to minify the code.

File: hawk/app/assets/javascripts/bigSlide.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+            view.toggleClose();
+          } else {
+            view.toggleOpen();
+          }
+        });
+
+        // this makes my eyes bleed, but adding it back in as it's a highly requested feature
+        if (settings.easyClose) {
+          $(document).on('click.bigSlide', function(e) {
+           if (!$(e.target).parents().andSelf().is(menuLink) && !$(e.target).closest(menu).length &&  controller.getState() === 'open')  {
```
Comment (tboerger) [inline hawk/app/assets/javascripts/bigSlide.js]: This is indeed not defined oO

File: hawk/app/assets/javascripts/bigSlide.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+      toggleClose: function() {
+        settings.beforeClose();
+        controller.changeState();
+        this.$menu.css(settings.side, '-' + this.width);
+        this.$push.css(settings.side, '0');
+        menuLink.removeClass(settings.activeBtn);
+        settings.afterClose();
+      }
+
+    }
```
Comment (tboerger) [inline hawk/app/assets/javascripts/bigSlide.js]: That should be added to be able to minify the code.

File: hawk/app/views/layouts/application.html.haml [hunk 7]
```diff
@@ -22,10 +22,8 @@
           = yield
 
           - if current_cib.sim?
-            .container-fluid
-              .row
-                .col-lg-12
-                  = render partial: "shared/simulator"
+            %nav#menu.panel{ role: "navigation" }
```
Comment (tboerger) [inline hawk/app/views/layouts/application.html.haml]: Do it, that's the haml lint rule.

File: hawk/app/views/layouts/withrightbar.html.haml [hunk 8]
```diff
@@ -22,7 +22,7 @@
           = yield
 
           - if current_cib.sim?
-            .container-fluid
+            %nav#menu.panel{ role: "navigation" }
```
Comment (tboerger) [inline hawk/app/views/layouts/withrightbar.html.haml]: Do it, that's the haml lint rule.

File: hawk/app/views/shared/_simulator.html.haml [hunk 9]
```diff
@@ -15,23 +15,22 @@
...
-              - unless current_cib.tickets.empty?
-                %button#sim-addticket.btn.btn-default.btn-sm{type: :button}
-                  = icon_text "plus", _("Ticket")
-              %button#sim-delete.btn.btn-default.btn-sm{type: :button}
-                = icon_tag "minus"
-            .col-sm-5
-              .well
-                %button#sim-results.btn.btn-default{type: :button, disabled: :disabled}
-                  = _("Results")
+            %select#sim-events.form-control{multiple: true}
```
Comment (tboerger) [inline hawk/app/views/shared/_simulator.html.haml]: Do it, that's the haml lint rule.

File: hawk/app/views/shared/_simulator.html.haml (hunk 1 above)
Comment (tboerger) [inline hawk/app/views/shared/_simulator.html.haml]: Do it, that's the haml lint rule.

**Comment by tboerger** [comment]
Please fix the lint errors before we review that.

File: hawk/app/views/shared/_simulator.html.haml (hunk 1 above)
Comment (thutterer) [inline hawk/app/views/shared/_simulator.html.haml]: Ok, after [looking it up](https://github.com/brigade/haml-lint/blob/master/lib/haml_lint/linter/README.md#classesbeforeids) it actually makes sense.

**Pull request closed without merging**